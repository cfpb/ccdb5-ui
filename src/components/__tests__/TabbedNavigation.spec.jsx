import configureMockStore from 'redux-mock-store'
import {
  mapDispatchToProps, mapStateToProps, TabbedNavigation
} from '../TabbedNavigation'
import { Provider } from 'react-redux'
import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import thunk from 'redux-thunk'

function setupSnapshot() {
  const middlewares = [ thunk ]
  const mockStore = configureMockStore( middlewares )
  const store = mockStore()

  return renderer.create(
    <Provider store={ store }>
      <TabbedNavigation />
    </Provider>
  )
}

describe( 'component: TabbedNavigation', () => {
  describe( 'initial state', () => {
    it( 'renders without crashing', () => {
      const target = setupSnapshot(false)
      let tree = target.toJSON()
      expect( tree ).toMatchSnapshot()
    } )

    it( 'shows the Trends tab', () => {
      const target = setupSnapshot(true)
      let tree = target.toJSON()
      expect( tree ).toMatchSnapshot()
    } )
  } )

  describe('buttons', () => {
    let cb = null
    let target = null

    beforeEach( () => {
      cb = jest.fn()
      target = shallow( <TabbedNavigation onTab={ cb } /> )
    } )

    it( 'tabChanged is called with Map when the button is clicked', () => {
      const prev = target.find( '.tabbed-navigation button.map' )
      prev.simulate( 'click' )
      expect( cb ).toHaveBeenCalledWith('Map')
    } )

    it( 'tabChanged is called with Trends when the button is clicked', () => {
      const prev = target.find( '.tabbed-navigation button.trends' )
      prev.simulate( 'click' )
      expect( cb ).toHaveBeenCalledWith('Trends')
    } )

    it( 'tabChanged is called with List when the button is clicked', () => {
      const prev = target.find( '.tabbed-navigation button.list' )
      prev.simulate( 'click' )
      expect( cb ).toHaveBeenCalledWith('List')
    } )
  })

  describe('mapDispatchToProps', () => {
    it('provides a way to call tabChanged', () => {
      const dispatch = jest.fn()
      mapDispatchToProps(dispatch).onTab()
      expect(dispatch.mock.calls.length).toEqual(1)
    })
  })

  describe( 'mapStateToProps', () => {
    it( 'maps state and props', () => {
      const state = {
        query: {
          tab: 'foo'
        }
      }
      let actual = mapStateToProps( state )
      expect( actual ).toEqual( { tab: 'foo' } )
    } )
  } )


} )
