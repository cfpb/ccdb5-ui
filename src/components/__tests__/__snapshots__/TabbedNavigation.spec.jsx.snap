// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: TabbedNavigation initial state renders without crashing 1`] = `
<div
  className="tabbed-navigation"
>
  <section>
    <button
      className="map tab"
      onClick={[Function]}
    >
      Map
    </button>
    <button
      className="trends tab"
      onClick={[Function]}
    >
      Trends
    </button>
    <button
      className="list tab"
      onClick={[Function]}
    >
      List
    </button>
  </section>
</div>
`;

exports[`component: TabbedNavigation initial state shows the Trends tab 1`] = `
<div
  className="tabbed-navigation"
>
  <section>
    <button
      className="map tab"
      onClick={[Function]}
    >
      Map
    </button>
    <button
      className="trends tab"
      onClick={[Function]}
    >
      Trends
    </button>
    <button
      className="list tab"
      onClick={[Function]}
    >
      List
    </button>
  </section>
</div>
`;
